### PR TITLE
human readable type indexer

### PR DESCRIPTION
**Pull request by geekscruff**
Moved indexing out of HumanReadableType into a HumanReadableTypeIndexer

### CODE REVIEW

File: spec/indexers/hyrax/human_readable_type_indexer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+RSpec.describe Hyrax::HumanReadableTypeIndexer do
+  subject(:solr_document) { service.to_solr }
+
+  let(:user) { create(:user) }
```
Comment (cjcolvar) [inline spec/indexers/hyrax/human_readable_type_indexer_spec.rb]: Do you need user?

File: spec/indexers/hyrax/human_readable_type_indexer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  let(:service) { described_class.new(resource: work) }
+  let(:work) { create_for_repository(:work) }
```
Comment (cjcolvar) [inline spec/indexers/hyrax/human_readable_type_indexer_spec.rb]: I think `build(:work)` would suffice here.

File: spec/indexers/hyrax/human_readable_type_indexer_spec.rb (hunk 1 above)
Comment (geekscruff) [reply spec/indexers/hyrax/human_readable_type_indexer_spec.rb]: You are right, no. I've changed this.

File: spec/indexers/hyrax/human_readable_type_indexer_spec.rb (hunk 2 above)
Comment (geekscruff) [reply spec/indexers/hyrax/human_readable_type_indexer_spec.rb]: Done.

**Pull request merged**